src/ngraph/ngraph_compiler.cc

```diff
@@ -109,16 +112,25 @@ void Compiler::Infer(const SimpleBindArg* simplebind) {
   }
 }
 
+static std::atomic<int> graph_counter(0);
+
+std::string get_ngraph_name() {
+  std::stringstream name;
+  name << "ngraph_" << std::this_thread::get_id() << "_" << graph_counter;
```

Comment:
This is not thread safe even with the `atomic` data type. You could do something like:
`name << "ngraph_" << std::this_thread::get_id() << "_" << graph_counter++;` and remove line 120.

Reply:
I also wonder if it's necessary to have thread id in the name if the `graph_counter` is already atomic and thread safe. The thread id will be potentially different from run to run.

Reply:
The main reason for the thread name is that sockeye dlopen's libmxnet multiple times in multiple threads, so with just a counter, it's not unique